float64 al
float64 ar